### ISSUE TITLE
Unable to add extra owners to application via terraform, but able to via Azure CLI

### ISSUE THREAD

**Issue by nxf5025** [hashicorp/terraform-provider-azuread#853]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version
Terraform v1.2.5
registry.terraform.io/hashicorp/azuread v2.26.1

### Affected Resource(s)

* `azuread_application`

### Terraform Configuration Files

```hcl
data "azuread_client_config" "current" {}

resource "azuread_application" "example" {
  display_name = "ExampleApplication"
  owners       = [data.azuread_client_config.current.object_id, "abcd1234-ef56-gh78-9100-12345abcdef"]
}
```

### Debug Output
https://gist.github.com/nxf5025/e376a719c7038725634079a2e7b22e94

### Expected Behavior
Application created with owner added.

### Actual Behavior
```
│ Error: Could not create application
│
│   with azuread_application.example,
│   on main.tf line 33, in resource "azuread_application" "example":
│   33: resource "azuread_application" "example" {
│
│ ApplicationsClient.BaseClient.Post(): unexpected status 403 with OData error: Authorization_RequestDenied: Insufficient privileges to complete the operation.
```

### Steps to Reproduce

1. Manually create Azure application and service principal and assign "Application.ReadWrite.OwnedBy"
2. Generate a client secret
3. Set the ARM_CLIENT_ID, ARM_CLIENT_SECRET, and ARM_TENANT_ID environment variables
4. Add an extra owner to the owners list in the terraform template
5. `terraform apply`

### Important Factoids
If I create the application without extra owners it all works as expected. I am then able to execute the following Azure CLI commands to add any additional owners without issue:

```
az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET -t $ARM_TENANT_ID
az ad app owner add --id <AppIDFromTerraform> --owner-object-id abcd1234-ef56-gh78-9100-12345abcdef
```

### References

* #0000

**Comment by Threpio** [comment]
Some actions with a service principle require two scopes to be assigned - "User Access Administrator" and "Contributo" - Can you check what your current configs permissions and scopes are?

**Comment by manicminer** [comment]
@nxf5025 Please can you try assigning the app role `Application.ReadWrite.All` to the principal you're using to run Terraform? Note that whilst we make a best effort to work as much as possible, due to API constraints we don't guarantee that all properties will work when assigning `Application.ReadWrite.OwnedBy` and YMMV depending on the combination of directory settings and privileges in use, as well as some other out of band factors.

**Comment by nxf5025** [comment]
Okay with the help from an Azure engineer we were able to dig into this a bit more. The Azure docs reference `Application.ReadWrite.All` is required for adding additional owners - https://docs.microsoft.com/en-us/graph/api/application-post-owners?view=graph-rest-1.0&tabs=http (Thanks @manicminer for pointing us in the right direction)

This conflicts with the fact that I'm able to add owners post application registration creation (The engineer is looking into this further)

However, to close this specific issue out I was able to add `User.Read.All` which then allowed me to add extra owners during creation.

So for now it seems like both `Application.ReadWrite.OwnedBy` and `User.Read.All` are required for this use case.
